### PR TITLE
fix: set default cost center as main onload and remove cost center colu…

### PR DESCRIPTION
**Pull request by hitenshKharva**
Ref Subtask 2: [Asana Task](https://app.asana.com/0/1199082161983365/1199966341817849)

https://user-images.githubusercontent.com/56474017/115190979-6c131e00-a106-11eb-92dc-d9b6016ad569.mp4

### CODE REVIEW

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js [hunk 1]
```diff
@@ -10,7 +10,18 @@ frappe.query_reports["Cost Center Financial Statements"] = {
 			"fieldtype": "Link",
 			"options": "Company",
 			"default": frappe.defaults.get_user_default("Company"),
-			"reqd": 1
+			"reqd": 1,
+			"on_change": function(query_report) {
+				var company = query_report.get_filter_value("company");
+				if (!company) {
+					return;
```
Comment (hrwX) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]: If company filter is empty, set cost center value as empty

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js [hunk 2]
```diff
@@ -72,6 +83,13 @@ frappe.query_reports["Cost Center Financial Statements"] = {
 			"fieldname": "cost_center",
 			"label": __("Cost Center"),
 			"fieldtype": "MultiSelectList",
+  			"default": function() {
+				frappe.model.with_doc("Company", frappe.defaults.get_user_default("Company"), function(r) {
```
Comment (hrwX) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]: - frappe.defaults.get_user_default("Company") save this in a variable

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js [hunk 3]
```diff
@@ -10,7 +10,18 @@ frappe.query_reports["Cost Center Financial Statements"] = {
 			"fieldtype": "Link",
 			"options": "Company",
 			"default": frappe.defaults.get_user_default("Company"),
-			"reqd": 1
+			"reqd": 1,
+			"on_change": function(query_report) {
+				var company = query_report.get_filter_value("company");
```
Comment (hrwX) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]:
```suggestion
				let company = query_report.get_filter_value("company");
```

**Review by hrwX — APPROVED** [review]

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js [hunk 4]
```diff
@@ -10,7 +10,18 @@ frappe.query_reports["Cost Center Financial Statements"] = {
...
+				if (!company) {
+					return;
+				}
+				frappe.model.with_doc("Company", company, function(r) {
```
Comment (hrwX) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]:
```suggestion

				frappe.model.with_doc("Company", company, function(r) {
```

Comment (hrwX) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]: I guess you get the document in the callback, you wont have to do a get_doc, please check that

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js (hunk 4 above)
Comment (hitenshKharva) [reply erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js]: frappe.model.with_doc returns the Company name in call back. get_doc is required to fetch the company cost center details

**Review by hrwX — DISMISSED** [review]

File: erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js [hunk 5]
```diff
@@ -72,6 +85,14 @@ frappe.query_reports["Cost Center Financial Statements"] = {
 			"fieldname": "cost_center",
 			"label": __("Cost Center"),
 			"fieldtype": "MultiSelectList",
+  			"default": function() {
+				let company = frappe.defaults.get_user_default("Company")
+				frappe.model.with_doc("Company", company, function(r) {
```
Comment (sahil28297) [inline erpnext/accounts/report/cost_center_financial_statements/cost_center_financial_statements.js:91]:
```suggestion
				if (!company) return;
				frappe.model.with_doc("Company", company, function(r) {
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**